lib/classes/Variables.js

```diff
@@ -308,7 +308,15 @@ class Variables {
     return BbPromise.all(populations).then((results) =>
       results.forEach((result) => {
         if (result.value !== result.populated) {
-          _.set(target, result.path, result.populated);
+          // We could not use direct `_.set` as due to security measures, it did not allow setting values
+          // if `prototype` was a part of the path and that caused an infinite loop. In our use-case, there are no
+          // security vulnerabilities.
+          const pathArray = Array.isArray(result.path) ? result.path : result.path.split('.');
+          if (pathArray.length > 1) {
+            _.get(target, pathArray.slice(0, -1))[pathArray.slice(-1)] = result.populated;
```

Comment:
I think it won't work if object for which property is destined doesn't already exists, e.g.

```javascript
obj = {};
_get(obj, ['one'])['two'] = 'value'; // Crash
```

Reply:
That's true - I thought about it and I believe in this particular case it's impossible because during assignment, we only deal with paths that are present and point to a value which has a variable that should be resolved. Am I missing something here and we can have a situation where we're resolving variable that will be assigned to a property that didn't exist before?

Reply:
Indeed, so maybe in that case let's simplify it to not rely on `_.get` (`_.get` has built in mechanism which ensures we will not crash if path doesn't exist, and here I believe we do not needed). e.g.

```javascript
const obj = target;
for (const property of path) obj = obj[property];
```

Reply:
great idea - I've adjusted my implementation